### PR TITLE
Allow changing password even with Discourse down

### PR DESCRIPTION
**Pull request by gberaudo**

### CODE REVIEW

**Comment by gberaudo** [comment]
As a side effect, make local development fun again!

**Comment by gberaudo** [comment]
As a reminder, the reason why we require Discourse to be up in other places is that we want to be sure the Discourse DB is kept synchronized with changes in the API. Since we use SSO and the passwords are only stored in the API, it is safe fo skip Discourse in this case.

File: c2corg_api/views/user.py [hunk 1]
```diff
@@ -187,11 +187,11 @@ def post(self):
                 r = client.redirect_without_nonce(user)
                 response['redirect_internal'] = r
             except:
-                # Any error with discourse must prevent login and validation
+                # Since only the password is changed, any error with discourse
+                # must not prevent login and validation.
```
Comment (tsauerwein) [inline c2corg_api/views/user.py]: If only the password is changed, could you skip the synchronization with Discourse in the first place?

Comment (gberaudo) [inline c2corg_api/views/user.py]: No, we want the user to be logged in at the end of the procedure.

Comment (tsauerwein) [inline c2corg_api/views/user.py]: Ok, I see. This is changing the password as part of the "requesting a new password" procedure.

**Comment by tsauerwein** [comment]
Feel free to merge.

**Comment by gberaudo** [comment]
Thanks for the review @tsauerwein.

**Pull request merged**